src/reducers/linter.spec.tsx

```diff
@@ -599,4 +601,78 @@ describe(__filename, () => {
       }).toThrow(/unknown types/);
     });
   });
+
+  describe('getMessagesForPath', () => {
+    const uid1 = 'global1';
+    const uid2 = 'global2';
+    const uid3 = 'line1-1';
+    const uid4 = 'line1-2';
+    const uid5 = 'line2';
+    const message1 = { line: null, uid: uid1 };
+    const message2 = { line: null, uid: uid2 };
+    const message3 = { line: 1, uid: uid3 };
+    const message4 = { line: 1, uid: uid4 };
+    const message5 = { line: 2, uid: uid5 };
+
+    it('throws an error if an extra key is found in the linter message map', () => {
+      const messages = createFakeLinterMessagesByPath({
+        messages: [{ line: null, uid: '123' }],
+      });
+
+      const unexpectedKey = 'future';
+      // Artifically inject a new key in the message map.
+      // @ts-ignore
+      messages[unexpectedKey] = {};
+
+      expect(() => {
+        getMessagesForPath(messages);
+      }).toThrow(new RegExp(`Unexpected key "${unexpectedKey}" found`));
+    });
+
+    it('aggregates global and byLine messages', () => {
+      const externalMessages = [
+        message1,
+        message2,
+        message3,
+        message4,
+        message5,
+      ];
+      const messages = createFakeLinterMessagesByPath({
+        messages: externalMessages,
+      });
+
+      const messagesForPath = getMessagesForPath(messages);
+      expect(messagesForPath.length).toEqual(5);
+      expect(messagesForPath[0]).toHaveProperty('uid', uid1);
+      expect(messagesForPath[1]).toHaveProperty('uid', uid2);
+      expect(messagesForPath[2]).toHaveProperty('uid', uid3);
+      expect(messagesForPath[3]).toHaveProperty('uid', uid4);
+      expect(messagesForPath[4]).toHaveProperty('uid', uid5);
+    });
+
+    it('returns with only global messages', () => {
+      const externalMessages = [message1, message2];
+      const messages = createFakeLinterMessagesByPath({
+        messages: externalMessages,
+      });
+
+      const messagesForPath = getMessagesForPath(messages);
+      expect(messagesForPath.length).toEqual(2);
+      expect(messagesForPath[0]).toHaveProperty('uid', uid1);
+      expect(messagesForPath[1]).toHaveProperty('uid', uid2);
+    });
+
+    it('returns with only byLine messages', () => {
+      const externalMessages = [message3, message4, message5];
```

Comment:
Same here. I think having variables for the UIDs is fine. As a reader, I'm trying to connect the UIDs so that's the hard part when the configuration is abstracted into variables.